storage/local/storage.go

```diff
@@ -360,11 +371,28 @@ func (s *memorySeriesStorage) GetMetricForFingerprint(fp clientmodel.Fingerprint
 
 // AppendSamples implements Storage.
 func (s *memorySeriesStorage) AppendSamples(samples clientmodel.Samples) {
+	if len(samples) == 0 {
```

Comment:
So I introduced the 0-len check when I introduced the OpenTSDB writing code in https://github.com/prometheus/prometheus/commit/61d26e84450fede401dc4dfb9abb51156279e390.

It seemed to matter only for something in the OpenTSDB path (probably it ended up sending empty messages to OpenTSDB), so I would remove it from here and only check in the OpenTSDB case instead. But even in that old commit, I can't immediately see why empty slices would have caused a problem.